src/context/meta-tags-context.php

```diff
@@ -655,6 +674,40 @@ public function generate_main_image_id() {
 		}
 	}
 
+	/**
+	 * Gets the main image ID for REST requests.
+	 *
+	 * @return int|null The main image ID.
+	 */
+	protected function get_main_image_id_for_rest_request() {
+		switch ( $this->page_type ) {
+			case 'Post_Type':
+				if ( $this->post instanceof WP_Post ) {
+					return $this->get_singular_post_image( $this->post->ID );
+				}
+				return null;
+			default:
+				return null;
+		}
+	}
+
+	/**
+	 * Gets the main image URL for REST requests.
+	 *
+	 * @return string|null The main image URL.
+	 */
+	private function get_main_image_url_for_rest_request() {
+		switch ( $this->page_type ) {
+			case 'Post_Type':
+				if ( $this->post instanceof WP_Post ) {
+					return $this->image->get_post_content_image( $this->post->ID );
```

Comment:
In the normal path, there is a check for the ID first to be able to use the attachment url by id. Can you not use that here too?

Reply:
I propose a different approach: we swap the order of the first two checks in `generate_main_image_url`, so that the `main_image_id` check happens regardless of the kind of request.